FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationResolvers/ConfigurationResolverV1.cs

```diff
@@ -15,8 +15,19 @@ public class ConfigurationResolverV1
             IServiceCollection services,
             IConfiguration configuration)
         {
-            services.Configure<FhirServerConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.FhirServerConfigurationKey).Bind(options));
+            // Try get DataSourceConfiguration first. If not found, create one from FhirServerConfiguration.
+            // If both exist, DataSourceConfiguration takes effect.
+            if (configuration.GetSection(ConfigurationConstants.DataSourceConfigurationKey).Exists())
+            {
+                services.Configure<DataSourceConfiguration>(options =>
+                    configuration.GetSection(ConfigurationConstants.DataSourceConfigurationKey).Bind(options));
+            }
+            else
+            {
```

Comment:
add commont to explain that it is just for backwards compatibility, so it would not be comsufe in the future.

Reply:
Sure. Comments are added before the `if` branch. I'll split them into different cases.